src/crimson/os/seastore/segment_manager/zns.cc

```diff
@@ -76,7 +81,7 @@ static zns_sm_metadata_t make_metadata(
     zone_capacity * zones_per_segment);
 
   zns_sm_metadata_t ret = zns_sm_metadata_t{
-    size,
+    total_size,
```

Comment:
I am sorry, I am not getting what I am missing here.
get_size() currently returns `metadata.size`, which is set by total_size, which is set as zone_capacity * segments.
get_segment_size() also returns zone_capacity(as zone capacity represents the actual writeable media available on the zone)

In ZNS device if there is a difference in zone capacity and zone size, then the entire device size is not useable.
For ex: If a ZNS SSD with 100 zones, each zone-size=2GB, but zone_capacity=1GB, then the device size is 200GB, but actual writeable media is only 100GB. In such cases if device size is set as 200GB, then garbage collection policies won't work as intended.  So I am setting the size as zone_capacity * segments(i.e. num-of-data-segments)

So `get_size()` in effect returns the same as `get_segment_size() * get_num_segments()`.
Is the suggestion just to remove the `size` variable

Reply:
Sorry if I'm not making myself clear enough. I'm following the difference between zone-size and zone-capacity.

I was meant that it looks better to use available_size(with zone capacity) directly instead of assigning it to total_size which has a different meaning.

For example:
```c++
size_t available_size = zone_capacity * segments;
assert(total_size == num_zones * zone_size);

WARN("Ignoring configuration values for device and segment size");
INFO(
  "device size {}, available {}, block_size {}, allocated_size {},"
  " total zones {}, zone_size {}, zone_capacity {},"
  " total segments {}, zones per segment {}, segment size {}",
  total_size,
  available_size,
  data.block_size,
  data.allocated_size,
  num_zones,
  zone_size,
  zone_capacity,
  segments,
  zones_per_segment,
  zone_capacity * zones_per_segment);

zns_sm_metadata_t ret = zns_sm_metadata_t{
  available_size,
  segment_size,
  zone_capacity * zones_per_segment,
  zones_per_segment,
  zone_capacity,
  data.block_size,
  segments,
  zone_size,
  zone_size * RESERVED_ZONES,
  meta};
return ret;
```

Does it look more straitforward?

Reply:
Ohh, I see what you mean. Will update this.